src/util/amount.rs

```diff
@@ -1375,4 +1449,47 @@ mod tests {
         let value_without: serde_json::Value = serde_json::from_str("{}").unwrap();
         assert_eq!(without, serde_json::from_value(value_without).unwrap());
     }
+
+    #[cfg(feature = "serde")]
+    #[test]
+    fn serde_as_sat_opt() {
+        use serde_json;
+
+        #[derive(Serialize, Deserialize, PartialEq, Debug, Eq)]
+        struct T {
+            #[serde(default, with = "::util::amount::serde::as_sat::opt")]
+            pub amt: Option<Amount>,
+            #[serde(default, with = "::util::amount::serde::as_sat::opt")]
+            pub samt: Option<SignedAmount>,
+        }
+
+        let with = T {
+            amt: Some(Amount::from_sat(2__500_000_00)),
+            samt: Some(SignedAmount::from_sat(-2__500_000_00)),
+        };
+        let without = T {
+            amt: None,
+            samt: None,
+        };
+
+        // Test Roundtripping
+        for s in [&with, &without].iter() {
+            let v = serde_json::to_string(s).unwrap();
+            let w : T = serde_json::from_str(&v).unwrap();
+            assert_eq!(w, **s);
+        }
+
+        let t: T = serde_json::from_str("{\"amt\": 250000000, \"samt\": -250000000}").unwrap();
+        assert_eq!(t, with);
+
+        let t: T = serde_json::from_str("{}").unwrap();
+        assert_eq!(t, without);
+
+        let value_with: serde_json::Value =
+            serde_json::from_str("{\"amt\": 250000000, \"samt\": -250000000}").unwrap();
+        assert_eq!(with, serde_json::from_value(value_with).unwrap());
+
+        let value_without: serde_json::Value = serde_json::from_str("{}").unwrap();
+        assert_eq!(without, serde_json::from_value(value_without).unwrap());
```

Comment:
Isn't this test redundant with the one on lines 1485-1486 or does the round-trip to `serde_json::Value` provide any more guarantees?

Reply:
No it isn't redundant -- serde_json serializes None as null rather than with `skip_serializing_if="Option::is_nonce"`. serde_json deserializes with the key present or not it seems. This is precisely where the error was previously; we were not trying with amt: null. It would be a breaking change to stop supporting "{}" now.

Reply:
What I fail to understand is the difference between:

- `"{}"` -> `T` (tested on line 1485)
- `"{}"` -> `serde_json::Value` -> `T` (tested on line 1492)

Reply:
Ah I thought you were referring to "roundtrip", which is the test we were previously lacking.

As to the redundancy of this test with the other I have nothing to say as it's just a copy of what we did previously.

Reply:
Fair enough. I agree that if then these simplifications should be handled in a separate PR. Maybe I'll get to that some time next week!

Reply:
That sounds like a great idea; serde_test sounds awesome :)